### ISSUE TITLE
Code in chapter headers can result in invalid html elements in ToC

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/mdBook#1372]
In the Rust Book chapter 15.1 it has the header

```markdown
## Using `Box<T>` to Point to Data on the Heap
```

This is rendered into the table of contents as

```html
<a href="ch15-01-box.html"><strong aria-hidden="true">15.1.</strong> Using Box<T> to Point to Data on the Heap</a>
```

Resulting in the `<T>` not being rendered as it is interpreted as an HTML element.

**Comment by ehuss** [comment]
This should be fixed via #1376.

**Comment by dtolnay** [comment]
Confirmed fixed.

![Screenshot from 2020-11-10 14-56-26](https://user-images.githubusercontent.com/1940490/98743467-f5a70e80-2364-11eb-9fc6-66905e4ec03e.png)
